components/style/values/specified/position.rs

```diff
@@ -232,9 +269,64 @@ impl ToComputedValue for Position {
 
     #[inline]
     fn to_computed_value(&self, context: &Context) -> computed_position::Position {
+        let horiz_keyword = self.horiz_keyword.unwrap_or(Keyword::Left);
+        let vert_keyword = self.vert_keyword.unwrap_or(Keyword::Top);
+
+        // Construct horizontal computed LengthOrPercentage
+        let horizontal = match horiz_keyword {
+            Keyword::Right => {
+                if let Some(x) = self.horiz_position {
+                    let (length, percentage) = match x {
+                        LengthOrPercentage::Percentage(Percentage(y)) => (None, Some(1.0 - y)),
+                        LengthOrPercentage::Length(y) => (Some(-y.to_computed_value(context)), Some(1.0)),
+                        _ => (None, None),
+                    };
+                    ComputedLengthOrPercentage::Calc(CalcLengthOrPercentage {
+                        length: length,
+                        percentage: percentage
+                    })
+                } else {
+                    ComputedLengthOrPercentage::Percentage(1.0)
+                }
+            },
+            Keyword::Center => {
+                horiz_keyword.to_length_or_percentage().to_computed_value(context)
```

Comment:
center is `50% + length`, so you need to take the length, and add 50% to its calc value percentage